client/src/components/Home/Home.js

```diff
@@ -1,19 +1,81 @@
-import React from "react";
-import Header from "../Header/Header";
+
+import React, { useEffect, useState } from "react";
+import './Search.css';
+import AnimeList from "../AnimeList/AnimeList";
+import { AnimeInfo } from "../AnimeList/AnimeInfo";
+import { AddToList } from "../AnimeList/AddToList";
+import { RemoveFromList } from "../AnimeList/RemoveFromList";
+import NavBar from "../Nav/Nav";
 function Home() {
-    return (
-        <>
-        <br></br>
-            <div className="Header">
-                <header className="App-header">
-                This is my home page
-                </header> 
 
+  const [search,setSearch]=useState('Naruto')
+  const [animeData,setAnimeData]=useState();
+  const [animeInfo,setAnimeInfo]=useState()
+  const [myAnimeList,setMyAnimeList]=useState([])
+
+  const addTo=(anime)=>{
+    const index=myAnimeList.findIndex((myanime)=>{
+        return myanime.mal_id === anime.mal_id
+    })
+    if(index < 0){
+      const newArray=[...myAnimeList,anime]
+      setMyAnimeList(newArray);
+    }
+    
+  }
+  const removeFrom=(anime)=>{
+    const newArray=myAnimeList.filter((myanime)=>{
+      return myanime.mal_id !== anime.mal_id
+    })
+    setMyAnimeList(newArray)
+  }
+  const getData=async()=>{
+      const res=await fetch(`https://api.jikan.moe/v4/anime?q=${search}&limit=20`)
+      const resData= await res.json();
+      setAnimeData(resData.data)
+  }
+  useEffect(()=>{
+    getData()
+  },[search])
+
+  return (
+    <>
+        <div className="header">
+            <NavBar/>
+          <div className="search-box">
+              <input type="search" placeholder="Search your anime" 
+              onChange={(e)=>setSearch(e.target.value)}/>
+          </div>
+        </div>
+
+        <div className="container">
+          <div className="animeInfo">
+           {animeInfo && <AnimeInfo animeInfo={animeInfo}/>}
+          </div>
+          <div className="anime-row">
+            <h2 className="text-heading">Anime</h2>
+            <div className="row">
+                <AnimeList 
+                animelist={animeData}
+                setAnimeInfo={setAnimeInfo}
+                animeComponent={AddToList}
```

Comment:
So I think what you're doing here passing a component in isn't a bad idea, but I think the hierarchy of information is a bit off. It seems like it might be clearer to pass an "animeAction" which is either "addAnime" or "removeAnime". When I see a generic "animeComponent" it seems pretty confusing.

I would bundle the functionality of the call back (either `addTo` or `removeFrom`) and the copy (e.g. the text from the `AddToList` or `RemoveFromList` components) into a single component.